### PR TITLE
SDKS-791: Fixing issue when writing splits on disk in app BG

### PR DESCRIPTION
**Pull request by javrudsky**
- Using concurrent classes
- Improved Split cache tests

### CODE REVIEW

File: src/test/java/io/split/android/engine/splits/SplitCacheTest.java [hunk 1]
```diff
@@ -108,5 +112,82 @@ public void testGetSplitNames() {
...
+
+        File rootFolder = new File(ROOT_FOLDER);
+        IStorage storage = new FileStorage(rootFolder, FOLDER);
+        //IStorage storage = new MemoryStorage();
+        CountDownLatch latch = new CountDownLatch(2);
+
+        for(int i = 0; i < 10000; i++) {
+            String splitName = "split-" + i;
+            try {
+                storage.write(FILE_PREFIX + splitName, String.format(JSON_SPLIT_TEMPLATE, splitName));
```
Comment (patricioe) [inline src/test/java/io/split/android/engine/splits/SplitCacheTest.java]: is this `.write` fails this test proceeds. Is that the expected behavior?

File: src/main/java/io/split/android/client/cache/SplitCache.java [hunk 2]
```diff
@@ -38,15 +41,15 @@
 
     private long mChangeNumber = -1;
     private Set<String> mRemovedSplits = null;
-    private Map<String, Split> mInMemorySplits = null;
-    private Map<String, Integer> mTrafficTypes = null;
+    private ConcurrentHashMap<String, Split> mInMemorySplits = null;
```
Comment (patricioe) [inline src/main/java/io/split/android/client/cache/SplitCache.java]: consider using Map and concurrentHashMap in the right side.

File: src/test/java/io/split/android/engine/splits/SplitCacheTest.java (hunk 1 above)
Comment (javrudsky) [reply src/test/java/io/split/android/engine/splits/SplitCacheTest.java]: Nice catch. I've removed. Thank you.

File: src/main/java/io/split/android/client/cache/SplitCache.java (hunk 2 above)
Comment (javrudsky) [reply src/main/java/io/split/android/client/cache/SplitCache.java]: My bad. Fixed. Thanks!

**Pull request merged**